Formula/postgresql@12.rb

```diff
@@ -81,22 +81,100 @@ def post_install
     return if ENV["CI"]
 
     (var/"log").mkpath
-    (var/"postgres").mkpath
-    unless File.exist? "#{var}/postgres/PG_VERSION"
-      system "#{bin}/initdb", "--locale=C", "-E", "UTF-8", "#{var}/postgres"
+    versioned_postgresql_data_dir.mkpath
+    unless versioned_data_dir_exists?
+      system "#{bin}/initdb", "--locale=C", "-E", "UTF-8", versioned_postgresql_data_dir
     end
   end
 
+  # Previous versions of this formula used the same data dir as the regular
+  # postgresql formula. So we check whether the versioned data dir exists
+  # and has a PG_VERSION file, which should indicate that the versioned
+  # data dir is in use. Otherwise, returns the old data dir path.
+  def postgresql_datadir
+    if versioned_data_dir_exists?
+      versioned_postgresql_data_dir
+    else
+      old_postgresql_data_dir
+    end
+  end
+
+  def versioned_postgresql_data_dir
+    "#{var}/#{name}"
+  end
+
+  def old_postgresql_data_dir
```

Comment:
```suggestion
  def old_postgres_data_dir
```

Reply:
Addressed in 983c4475a63bd4e23dfabce8a2ae4cfc1e2d0734